src/cdk/table/sticky-styler.ts

```diff
@@ -188,37 +188,38 @@ export class StickyStyler {
       return;
     }
 
-    // If positioning the rows to the bottom, reverse their order when evaluating the sticky
-    // position such that the last row stuck will be "bottom: 0px" and so on. Note that the
-    // sticky states need to be reversed as well.
-    const rows = position === 'bottom' ? rowsToStick.slice().reverse() : rowsToStick;
-    const states = position === 'bottom' ? stickyStates.slice().reverse() : stickyStates;
-
-    // Measure row heights all at once before adding sticky styles to reduce layout thrashing.
-    const stickyOffsets: number[] = [];
-    const stickyCellHeights: (number | undefined)[] = [];
-    const elementsToStick: HTMLElement[][] = [];
-    for (let rowIndex = 0, stickyOffset = 0; rowIndex < rows.length; rowIndex++) {
-      if (!states[rowIndex]) {
-        continue;
-      }
+    // Coalesce with other sticky row updates (top/bottom), sticky columns updates
+    // (and potentially other changes like column resize).
+    this._coalescedStyleScheduler.schedule(() => {
```

Comment:
This is a pretty big diff, but I just moved the measurements into the `schedule` call.